### PR TITLE
REF: dont pass keys through wrap_applied_output

### PR DESCRIPTION
**Pull request by jbrockmendel**
This makes it easier to reason about these `keys`, which I'm hopeful will lead to getting rid of them entirely and just using result_index.

### CODE REVIEW

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -726,10 +726,15 @@ def _get_group_keys(self):
             return get_flattened_list(ids, ngroups, self.levels, self.codes)
 
     @final
-    def apply(self, f: F, data: FrameOrSeries, axis: int = 0):
+    @cache_readonly
+    def group_keys_seq(self):
```
Comment (jreback) [inline pandas/core/groupby/ops.py:730]: why are you adding this?

Comment (jbrockmendel) [reply pandas/core/groupby/ops.py:730]: without adding this, the PR makes us call _get_group_keys repeatedly; this is a cached version of that call

Comment (jbrockmendel) [reply pandas/core/groupby/ops.py]: updated to just remove _get_group_keys and only have the cached version

**Comment by jreback** [comment]
sure

**Pull request merged**